### PR TITLE
Refactor into loadable class

### PR DESCRIPTION
**Pull request by pieterza**
I tried to keep the CLI behavior identical to original, but we are now also able to load the Python 
`SlackClient` class directly into other python projects.

Requires complete testing to ensure I did not  break cli behavior
Some imports can be moved and may still be duplicated, will get on that after we know it's fully backwards compatible.

Example:

* Native code usage:

```
from ritdu_slacker.api import SlackClient
client = SlackClient()
client.post_message("via python api","ringier-southafrica","#pe-alerts")
{'message_uuid': '9890b802-fac3-4e61-bbe8-b53cc17fc581', 'message_ts': '1677473299.255969', 'thread_uuid': '9890b802-fac3-4e61-bbe8-b53cc17fc581', 'thread_ts': '1677473299.255969', 'channel': 'CV3JFH08J'}
```

* CLI usage:

`ritdu-slacker  message --text "Hello from cli"  --workspace ringier-southafrica  -c "#pe-alerts"`
`ritdu-slacker file  --workspace ringier-southafrica  -c @pieters -f  ~/list`
`ritdu-slacker  message --text "Hello from cli"  --workspace ringier-southafrica  -c "#pe-alerts"  --thread-uuid 9890b905-7442-470d-b161-fb76e410704a`

Please give it a quick test @puzzledvacuum if you find a gap, but seems to me like it's good and non-breaking in compatibility.

### CODE REVIEW

File: ritdu_slacker/cli.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        text,
+        thread_uuid,
+        message_uuid,
+        message_or_thread_uuid,
+        workspace,
+        channel,
+        thread_broadcast,
+    ):
+        sender = SlackClient()
+        result = sender.post_message(
```
Comment (pieterza) [inline ritdu_slacker/cli.py:68]: This should ideally be `self.sender.post_message(, but I ran into trouble.

**Review by ginahorscroft — APPROVED** [review]

**Comment by puzzledvacuum** [comment]
@pieterza Could you use `black` to `fmt` the python please?

**Comment by puzzledvacuum** [comment]
@pieterza Looks like we lost some functionality?
```bash
$ ritdu-slacker --version                                                                    
Usage: ritdu-slacker [OPTIONS] COMMAND [ARGS]...

Error: No such option: --version
```

**Comment by pieterza** [comment]
> @pieterza Could you use `black` to `fmt` the python please?

Done

**Comment by pieterza** [comment]
> @pieterza Looks like we lost some functionality?
> 
> ```shell
> $ ritdu-slacker --version                                                                    
> Usage: ritdu-slacker [OPTIONS] COMMAND [ARGS]...
> 
> Error: No such option: --version
> ```

Will handle, tnx!

**Review by puzzledvacuum — APPROVED** [review]
:excellent: 💎

**Pull request merged**